### PR TITLE
Add support for efficient preprocessing of DASH manifests after parsing into an XMLDocument

### PR DESCRIPTION
**Pull request by caridley**
## Description
Implements #3339

Added support for efficient preprocessing of DASH manifest after they have been parsed into an XMLDocument

This is configure with the new manifest.dash.manifestPreprocessor  setting.

This is need to efficiently repair manifests that are not compatible with Shaka player.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

Documentation for DashManifestConfiguration changes

## Checklist:

- [x] I have signed the Google CLA <https://cla.developers.google.com>
- [x] My code follows the style guidelines of this project
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have verified my change on multiple browsers on different platforms
- [x] I have run `./build/all.py` and the build passes
- [x] I have run `./build/test.py` and all tests pass

### CODE REVIEW

File: test/dash/dash_parser_manifest_unit.js [hunk 1]
```diff
@@ -1790,6 +1790,48 @@ describe('DashParser Manifest', () => {
...
+      '      </Representation>',
+      '    </AdaptationSet>',
+      '  </Period>',
+      '</MPD>',
+    ].join('\n');
+
+    fakeNetEngine.setResponseText('dummy://foo', manifestText);
+    const config = shaka.util.PlayerConfiguration.createDefault().manifest;
+    config.dash.manifestPreprocessor = (mpd) => {
+      const selector = 'AdaptationSet[mimeType="text/vtt"';
```
Comment (joeyparrish) [inline test/dash/dash_parser_manifest_unit.js:1821]: Is this missing a closing bracket?

Comment (joeyparrish) [reply test/dash/dash_parser_manifest_unit.js:1821]: Looks like it works without it.  I learn something new every day!

**Review by joeyparrish — APPROVED** [review]

**Comment by caridley** [comment]
@joeyparrish can this be cherry picked into v3.0.x and v3.1.x ?

**Comment by joeyparrish** [comment]
Since v3.0.0, we are following [Semantic Versioning](https://semver.org/) closely.  This means all new features require a minor version bump, and can't be cherry-picked into older release branches.

**Pull request merged**